src/bionics_ui.cpp

```diff
@@ -36,6 +37,94 @@ enum bionic_menu_mode {
     EXAMINING,
     REASSIGNING
 };
+
+enum bionic_sort_mode {
+    POWER,
+    NAME,
+};
+bionic_sort_mode _bionic_sort_mode = bionic_sort_mode::POWER;
+
+auto sort_mode_str( bionic_sort_mode mode ) -> std::string
+{
+    std::string ret;
+    switch( mode ) {
+        case bionic_sort_mode::POWER:
+            ret = _( "Power Usage" );
+            break;
+        case bionic_sort_mode::NAME:
+            ret = _( "Name" );
+            break;
+    }
+    return ret;
+}
+
+auto is_power_cbm( const bionic_data &data ) -> bool
+{
+    return !data.fuel_opts.empty() || data.is_remote_fueled;
+}
+
+struct bionic_sort_less {
+    auto operator()( const bionic *lhs, const bionic *rhs ) const -> bool {
+        bool less = false;
+        const bionic_data &lbd = lhs->info();
+        const bionic_data &rbd = rhs->info();
+
+        switch( _bionic_sort_mode ) {
+            case bionic_sort_mode::POWER: {
+                units::energy lbd_sort_power = is_power_cbm( lbd ) ? units::energy( -1_kJ * lbd.fuel_efficiency ) :
+                                               lbd.power_activate + lbd.power_over_time;
+                units::energy rbd_sort_power = is_power_cbm( rbd ) ? units::energy( -1_kJ * rbd.fuel_efficiency ) :
+                                               rbd.power_activate + rbd.power_over_time;
+                if( lbd_sort_power != rbd_sort_power ) {
+                    less = lbd_sort_power < rbd_sort_power;
+                    break;
+                }
+            }
+            /* fallthrough */
+            case bionic_sort_mode::NAME:
+                less = lbd.name.translated() < rbd.name.translated();
```

Comment:
```suggestion
                return localized_compare( lbd.name.translated(), rbd.name.translated() );
```

Reply:
I *hate* having return statements all over the place. There should be only one per function except in very odd circumstances

Reply:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rnr-single-return

Reply:
That looks like a bad opinion piece. For example it adds an `else res="zero"` to make a point. but that should have been in the variable declaration already, etc.

Reply:
Also, in this case it doesn't change the number of lines of code in any way

Reply:
I would say if the structure of the method is clear enough there's no need to stick to a single return statement. What's problematic of multiple return statements is IMO when some of the return statements are buried deep in the middle of the code, it becomes hard to analyse the code, which is not the case here.

Reply:
Alright, I will compromise. Thanks